demos/demo_fiwalk_diskimage.py

```diff
@@ -0,0 +1,26 @@
+import sys
+import io
+from dfxml import fiwalk
+
+def writeDfxml(imageFile, outFile):
+    """Generate filesystem metadata for disk image and and write resulting dfxml to file"""
+
+    # Analyse image file
+    with open(imageFile, "rb") as ifs:
+        fwOutBuffer = fiwalk.fiwalk_xml_stream(imagefile=ifs)
+        fwOut = fwOutBuffer.read()
+
+    # Write dfxml to output file
+    with io.open(outFile, "wb") as fOut:
+        fOut.write(fwOut)
```

Comment:
Thanks for the contribution. My concern about this is that it requires the entire XML stream to be buffered in RAM at once (in `fwOut`) but there may be no realistic way to do this. It will fail for large images, however, unless run on systems with a huge amount of available memory. (The XML format is not very efficient.)

Reply:
If there's a more efficient way to do this I'm all for it, because yes I can see how this could easily go wrong for large images. I've never worked with io.BufferedReader objects before though, and I'm not quite sure how to do this.

Reply:
I don't think this changes the status quo for how DFXML has been implemented to date.  The read-write library `dfxml/objects.py` does some full-in-memory string construction as well.

My litmus test (not to include in CI!) would probably be seeing if this can run against the NPS 2TB image.

Reply:
It’s complicated. I believe there is an iterator in the dfxml package that runs fiwalk as a sub process , parses it with Expat, and calls a provided call back with an object as the argument. This is the interface that all of my code used. But that was 10 years ago.